### PR TITLE
LPS-163586 Add the UI for the Preview Action to the Version History for Web Content

### PR DESCRIPTION
**Pull request by beltranrengifo**
[Jira task](https://issues.liferay.com/browse/LPS-163586)
[Jira story](https://issues.liferay.com/browse/LPS-161020)

## Motivation

As a Content Marketer, I want to preview versions of a Web Content, so that I can easily review these versions from the Content Dashboard.

## Proposal

We are adding the barebones for the new actions dropdown for each version item.

## How to verify

~~Create some web content or document, generate several versions and visit the Content Dashboard Info Panel.~~

Currently there's no way of verify this FE implementation, as there's no exposed object from the BE. However you can simply fake props by using:

```
allVersions,
formatDate,
languageTag = 'en',
versionActions = [
		{
			action: 'preview',
			actionLabel: 'Preview',
			icon: 'view',
			previewURL:
				// eslint-disable-next-line @liferay/portal/no-localhost-reference
				'http://localhost:8080/home?p_p_id=com_liferay_journal_web_portlet_JournalPortlet&p_p_lifecycle=0&p_p_state=pop_up&p_p_mode=view&refererPlid=1&_com_liferay_journal_web_portlet_JournalPortlet_mvcPath=%2Fpreview_article_content.jsp&_com_liferay_journal_web_portlet_JournalPortlet_groupId=20121&_com_liferay_journal_web_portlet_JournalPortlet_articleId=43137&_com_liferay_journal_web_portlet_JournalPortlet_version=2.9&p_p_auth=0g9W4qxP',
			title: 'Preview',
			useModal: true,
		},
	],
```

as props for the `VersionsContent` component.

image

### CODE REVIEW

**Comment by beltranrengifo** [comment]
hey @cgoncas @jkappler @claraizquierdo 

I see the current implementation for the preview version action generates an action button [here](https://github.com/liferay/liferay-portal/blob/b708db5d9946c5a91a7973c16958c06bd4ffa9f9/modules/apps/journal/journal-web/src/main/java/com/liferay/journal/web/internal/servlet/taglib/util/JournalArticleActionDropdownItemsProvider.java#L688), and looks like this:

```
<button
  data-previewurl="http://localhost:8080/home?p_p_id=com_liferay_journal_web_portlet_JournalPortlet&amp;p_p_lifecycle=0&amp;p_p_state=pop_up&amp;p_p_mode=view&amp;refererPlid=1&amp;_com_liferay_journal_web_portlet_JournalPortlet_mvcPath=%2Fpreview_article_content.jsp&amp;_com_liferay_journal_web_portlet_JournalPortlet_groupId=20121&amp;_com_liferay_journal_web_portlet_JournalPortlet_articleId=43137&amp;_com_liferay_journal_web_portlet_JournalPortlet_version=2.9&amp;p_p_auth=awfItc30"
  data-action="preview"
  data-title="Web content for the summer" 
  icon="view" 
  class="dropdown-item" 
  role="menuitem">
    <span class="dropdown-item-indicator-start">
      <svg class="lexicon-icon lexicon-icon-view" role="presentation">
        <use xlink:href="http://localhost:8080/o/admin-theme/images/clay/icons.svg#view"></use>
      </svg>
    </span>
    Preview
</button>
```
Then the FE uses `openModal` util to launch the modal, [here](https://github.com/liferay/liferay-portal/blob/9e101809e2d42005e83befcb0faa0fd783fc2016/modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/ElementsDefaultPropsTransformer.js#L122).
I think we can follow the same approach, maybe use the same branding:
- previewurl or previewURL, better: the content url (just the content, the title goes other way)
- action: the action unique identification
- title: the web content title in this version
- icon

I'm going to prepare a quick POC, plz let me know how this sounds to you.
Thanks a lot!

**Comment by beltranrengifo** [comment]
ci:test:sf

**Comment by beltranrengifo** [comment]
ci:test:sf

**Comment by beltranrengifo** [comment]
ci:test:relevant

**Comment by jkappler** [comment]
ci:test:relevant

File: modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/SidebarPanelInfoView/VersionActions.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+interface IProps {
+	actions: IAction[];
+}
+
+interface IAction {
+	action: string;
+	actionLabel: string;
+	icon?: string;
+	previewURL: string;
```
Comment (claraizquierdo) [inline modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/SidebarPanelInfoView/VersionActions.tsx:80]: Hi @beltranrengifo , I prefer to use a more generic name for the url not including the name of the action. Something like actionURL could be better. What do you think?

Comment (beltranrengifo) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/SidebarPanelInfoView/VersionActions.tsx:80]: You're totally right @claraizquierdo 
I think I'll also remove the `useModal` until it's needed, which maybe will never occur...
What do you think?

Comment (claraizquierdo) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/SidebarPanelInfoView/VersionActions.tsx]: I agree with you @beltranrengifo . Remove it until it is needed.

**Comment by beltranrengifo** [comment]
@claraizquierdo 
fixed, rebased and push -f

**Comment by beltranrengifo** [comment]
ci:test:relevant

**Comment by beltranrengifo** [comment]
ci:test:relevant

File: modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/SidebarPanelInfoView/VersionActions.tsx (hunk 1 above)
Comment (beltranrengifo) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/SidebarPanelInfoView/VersionActions.tsx]: > I agree with you @beltranrengifo . Remove it until it is needed.

Perfect! So do you think we can ship this part?

**Review by claraizquierdo — APPROVED** [review]

**Comment by beltranrengifo** [comment]
ci:forward

**Comment by beltranrengifo** [comment]
ci:forward

**Comment by beltranrengifo** [comment]
ci:test:sf

**Comment by beltranrengifo** [comment]
ci:test:relevant

**Comment by beltranrengifo** [comment]
ci:forward

**Pull request closed without merging**